docs/csharp/csharp-6.md

```diff
@@ -1,17 +1,570 @@
+location of the original throw point has been lost. 
+
+Contrast that with how the code using an exception filter is processed:
+The exception filter expression evaluates to `false`. Execution never
+enters the `catch` clause. No stack unwinding takes place. The original
+throw location is preserved for any debugging activities that would take
+place later.
+
+Whenever you need to evaluate fields or properties of an exception, instead
+of relying on solely on the exception type, use an exception filter to
+preserve more debugging information.
+
+Another recommended pattern with exception filters is to use them for
+logging routines. This usage also leverages the manner in which the exception
+throw point is preserved when an exception filter evaluates to `false`.
+
+A logging method would be a method whose argument is the exception that
+unconditionally returns `false`:
+
+[!code-csharp[ExceptionFilterLogging](../../samples/snippets/csharp/new-in-6/ExceptionFilterHelpers.cs#ExceptionFilterLogging)]
+
+Whenever you want to log an exception, you can add a catch clause, and
+use this method as the exception filter:
+
+[!code-csharp[LogException](../../samples/snippets/csharp/new-in-6/program.cs#LogException)]
+
+The exceptions are never caught, because the `LogException` method always
+returns `false`. That always false exception filte means that you can
+place this logging handler before any other exception handlers:
+
+[!code-csharp[LogExceptionRecovery](../../samples/snippets/csharp/new-in-6/program.cs#LogExceptionRecovery)]
+
+The above example highlights a very important facet of exception filters.
+The exception filters enable scenarios where a more general exception
+catch clause may appear before a more specific one. It's also possible
+to have the same exception type appear in multiple catch clauses:
+
+[!code-csharp[HandleNotChanged](../../samples/snippets/csharp/new-in-6/NetworkClient.cs#HandleNotChanged)]
+
+Another recommended pattern helps prevent catch clauses from processing
+exceptions when a debugger is attached. This technique enables you to
+run an application with the debugger, and stop execution when an exception
+is thrown.
+
+In the code, add an exception filter so that any recovery code executes
+only when a debugger is not attached:
+
+[!code-csharp[LogExceptionDebugger](../../samples/snippets/csharp/new-in-6/program.cs#LogExceptionDebugger)]
+
+After adding this in code, you set your debugger to break on all unhandled
+exceptions. Run the program under the debugger, and the debugger breaks
+whenever `PerformFailingOperation()` throws a `RecoverableException`.
+The debugger breaks your program, because the catch clause won't be executed
+due to the false-returning exception filter.
+
+## `nameof` Expressions
+
+The `nameof` expression evaluates to the name of a symbol. It's a great
+way to get tools working with you whenever you need the name of a variable,
```

Comment:
"great way to get tools working with you" - I don't understand what you are trying to convey here.